sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -1605,4 +1625,292 @@ public String generateSas(BlobServiceSasSignatureValues blobServiceSasSignatureV
             getSnapshotId(), getVersionId())
             .generateSas(SasImplUtils.extractSharedKeyCredential(getHttpPipeline()));
     }
+
+    /* TODO (gapra): Quick Query service docs. */
+    /**
+     * Queries the entire blob.
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/">Azure Docs</a></p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.specialized.BlobAsyncClientBase.query#String}
+     *
+     * @param expression The query expression.
+     * @return A reactive response containing the queried data.
+     */
+    public Flux<ByteBuffer> query(String expression) {
```

Comment:
I put the Query API on BlockBlobClient, since it currently only works with block blobs.  If other blobs, we can move it to BaseBlobClient (assuming you have a BaseBlobClient in Java)

Reply:
I had to do this since you can't get a snapshot block blob client. We figured this is fine since there are some other APIs on Blob Base that don't work for specific types of blobs